apps/full-site-editing/full-site-editing-plugin/site-editor-experiment/index.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Site Editor experiment file.
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE;
+
+/**
+ * Enables/Disables site editor experiment per blog sticker.
+ */
+function conditionally_enable_site_editor() {
+	// Check blog sticker for access to Site Editor.
+	if ( ! has_blog_sticker( 'core_site_editor_enabled' ) ) {
+
+		// Check if experiment is enabled: disable if needed.
+		if ( gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) ) {
+			set_site_editor_experiment( '0' );
+		}
+		return;
+	}
+
+	// Check if experiment is disabled: enable if needed.
+	if ( ! gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) ) {
+		set_site_editor_experiment( '1' );
+	}
+
+	add_site_editor_plugin_page();
+	override_site_editor_init();
+}
+
+/**
+ * Sets the Site Editor experiment on or off.
+ *
+ * @param {string} $value - '0' for OFF, '1' for ON.
+ */
+function set_site_editor_experiment( $value ) {
+	$options = get_option( 'gutenberg-experiments' );
+
+	// Initialize options array if needed (may be set to false by default on dotcom).
+	if ( ! $options ) {
+		$options = array();
+	}
+	$options['gutenberg-full-site-editing']      = $value;
+	$options['gutenberg-full-site-editing-demo'] = $value;
+	update_option( 'gutenberg-experiments', $options );
+}
+
+/**
+ * Adds Site Editor page to wp-admin sidebar.
+ */
+function add_site_editor_plugin_page() {
+	// Without overriding init, this would need to be a submenu of gutenberg.
+	add_menu_page(
+		__( 'Site Editor (beta)', 'gutenberg' ),
+		__( 'Site Editor (beta)', 'gutenberg' ),
+		'edit_theme_options',
+		'gutenberg-edit-site',
+		'gutenberg_edit_site_page',
+		'dashicons-edit'
+	);
+}
+
+/**
+ * Initialize the Gutenberg Edit Site Page.
+ * Copy of core function from 7.3.0 - gutenberg_edit_site_init
+ *
+ * Currently overriding to change hook requirement to allow Site Editor at top level sidebar.
+ * And to troubleshoot the wp.editSite.initialize CORS error.
+ *
+ * Any changes will be prefixed with @DOTCOM_CUSTOMIZATION comment.
+ *
+ * @param string $hook Page.
+ */
+function fse_plugin_edit_site_init( $hook ) {
+	global $_wp_current_template_id;
+	// @DOTCOM_CUSTOMIZATION - rename hook to allow toplevel
+	if ( 'toplevel_page_gutenberg-edit-site' !== $hook ) {
```

Comment:
1 from above - Change the $hook accepted by the early return statement to allow our 'Site Editor' button to be at the top-level of the sidebar.